### PR TITLE
:sparkles: Add nunchuck driver

### PR DESCRIPTION
**Pull request by MaliaLabor**

### CODE REVIEW

File: include/libhal-nunchuck/nunchuck.hpp [hunk 1]
```diff
@@ -14,7 +14,83 @@
...
+  };
+
+  /**
+   * @brief Create a nunchuck object
+   * @param p_i2c - i2c object that the nunchuck will use
+   */
+  nunchuck(hal::i2c& p_i2c);
+
+private:
+  gamepad_data driver_read();
```
Comment (kammce) [inline include/libhal-nunchuck/nunchuck.hpp]:
```suggestion

  gamepad_data read();
private:
```
Should make this public so the user can call it to get the gamepad data. We can also reduce the name to just `read`. `driver_read` would be used if this was implementing an interface.

File: include/libhal-nunchuck/nunchuck.hpp [hunk 2]
```diff
@@ -14,7 +14,83 @@
...
+    /**
+     * @brief Get z-axis data of the accelerometer sensor.
+     * @return uint16_t representing the z-axis data of the accelerometer.
+     * Center position is about 512. Reducing values means downward movement,
+     * increasing values means upward movement.
+     */
+    std::uint16_t accelerometer_z();
+    /**
+     * @brief Get z button pressed state.
+     * @return Boolean representing if the z button is pressed or not.
```
Comment (kammce) [inline include/libhal-nunchuck/nunchuck.hpp]:
```suggestion
     * @return true representing if the z button is pressed
     * @return false representing if the z button is unpressed
```

File: include/libhal-nunchuck/nunchuck.hpp [hunk 3]
```diff
@@ -14,7 +14,83 @@
...
+     */
+    bool z_button();
+    /**
+     * @brief Get c button pressed state.
+     * @return Boolean representing if the c button is pressed or not.
```
Comment (kammce) [inline include/libhal-nunchuck/nunchuck.hpp]: Same as z_button.

File: include/libhal-nunchuck/nunchuck.hpp [hunk 4]
```diff
@@ -14,7 +14,83 @@
...
+     */
+    bool c_button();
+  };
+
+  /**
+   * @brief Create a nunchuck object
+   * @param p_i2c - i2c object that the nunchuck will use
```
Comment (kammce) [inline include/libhal-nunchuck/nunchuck.hpp]:
```suggestion
   * @param p_i2c - the driver for the i2c bus the nunchuck is connected to.
```

**Review by kammce — CHANGES_REQUESTED** [review]

File: include/libhal-nunchuck/nunchuck.hpp (hunk 1 above)
Comment (MaliaLabor) [reply include/libhal-nunchuck/nunchuck.hpp]: done

File: include/libhal-nunchuck/nunchuck.hpp (hunk 2 above)
Comment (MaliaLabor) [reply include/libhal-nunchuck/nunchuck.hpp]: Changed return documentation

File: include/libhal-nunchuck/nunchuck.hpp (hunk 3 above)
Comment (MaliaLabor) [reply include/libhal-nunchuck/nunchuck.hpp]: Changed return documentation

File: include/libhal-nunchuck/nunchuck.hpp (hunk 4 above)
Comment (MaliaLabor) [reply include/libhal-nunchuck/nunchuck.hpp]: Changed parameter documentation

**Review by kammce — APPROVED** [review]

**Pull request merged**